ambry-rest/src/main/java/com/github/ambry/rest/NettyResponseChannel.java

```diff
@@ -681,10 +706,16 @@ private HttpResponseStatus getHttpResponseStatus(ResponseStatus responseStatus)
    * </p>
    * May also close the channel if the class internally is forcing a close (i.e. if {@link #close()} is called.
    * @param closeNetworkChannel network channel is closed if {@code true}.
+   * @param shouldDelay network channel may be closed after certain delay if {@code true}.
    */
-  private void completeRequest(boolean closeNetworkChannel) {
+  private void completeRequest(boolean closeNetworkChannel, boolean shouldDelay) {
     if ((closeNetworkChannel || forceClose) && ctx.channel().isOpen()) {
-      writeFuture.addListener(ChannelFutureListener.CLOSE);
+      if (!shouldDelay || (request != null && !request.getRestMethod().equals(RestMethod.POST)) || (this.nettyConfig.nettyServerCloseDelayTimeoutMs == 0)) {
```

Comment:
Need to format this class for line wrapping.  In LI Intellij, do Code/Reformat Code or key shortcut Cmd-Alt-L

Reply:
also it's a bit hard to read the line, I suggest reversing this line and changing the OR's to AND's and removing NOT's (meaning we do want to delay the close).

Reply:
Done. Thanks much for the formatting help.